src/spooq2/transformer/flattener.py

```diff
@@ -0,0 +1,328 @@
+        with specific suffixes (_at, _time, _date)
+
+    ignore_ambiguous_columns : :any:`bool`, Defaults to True
+        This flag is forwarded to the Mapper Transformer.
+    """
+
+    def __init__(self, pretty_names=True, keep_original_columns=False, convert_timestamps=True, ignore_ambiguous_columns=True):
+        super().__init__()
+        self.pretty_names = pretty_names
+        self.keep_original_columns = keep_original_columns
+        self.convert_timestamps = convert_timestamps
+        self.ignore_ambiguous_columns = ignore_ambiguous_columns
+        self.python_script = []
+
+    def transform(self, input_df):
+        exploded_df, mapping = self._explode_and_get_mapping(input_df)
+        mapped_df = Mapper(mapping=mapping, ignore_ambiguous_columns=self.ignore_ambiguous_columns).transform(exploded_df)
+        return mapped_df
+
+    def get_script(self, input_df):
+        """
+        Flattens and explodes an input DataFrame but instead of directly applying it to the input DataFrame,
+        it returns a script that contains all imports, explosion transformations, and the resulting mapping.
+
+        Parameters
+        ----------
+        input_df : :py:class:`pyspark.sql.DataFrame`
+            Input DataFrame
+
+        Returns
+        -------
+        :any:`list` of :any:`str`
+            List of code lines to execute the flattening steps explicitly.
+
+        Hint
+        ----
+        See :py:mod:`spooq2.transformer.flattener.Flattener.export_script` for exporting the script to a file.
+        """
+        self._explode_and_get_mapping(input_df)
+        return self.python_script
+
+    def export_script(self, input_df, file_name):
+        """
+        Flattens and explodes an input DataFrame but instead of directly applying it to the input DataFrame,
+        it exports a script that contains all imports, explosion transformations, and the resulting mapping
+        to an external file.
+
+        Parameters
+        ----------
+        input_df : :py:class:`pyspark.sql.DataFrame`
+            Input DataFrame
+
+        file_name : :any:`str`
+            Name of file to which the script should be exported to.
+
+        Attention
+        ---------
+        Currently, only local paths are possible as it uses Python's internal ``open()`` function.
+
+        Hint
+        ----
+        See :py:mod:`spooq2.transformer.flattener.Flattener.get_script` for getting the script as a python object.
+        """
+        with open(file_name, "w") as file_handle:
+            for line in self.get_script(input_df):
+                file_handle.write(f"{line}\n")
+        self.logger.info(f"Flattening Tranformation Script was exported to {file_name}")
+
+    def _explode_and_get_mapping(self, input_df):
+        self.logger.info("Exploding Input DataFrame and Generating Mapping (This can take some time depending on the complexity of the input DataFrame)")
+        initial_mapping = []
```

Comment:
no need to declare initial_mapping inside this function, only used inside _get_preliminary_mapping and is always an empty list
same for current_path and exploded_arrays